index.bs

```diff
@@ -4053,164 +4253,973 @@ To <dfn>get the response data</dfn> given |response|:
    field set to |headers|, the <code>mimeType</code> field set to |mime type|, the
    <code>bytesReceived</code> field set to |bytes received|, the
    <code>headersSize</code> field set to |headers size|, the
-   <code>bodySize</code> field set to |body size|, and the <code>content</code>
-   field set to |content|.
+   <code>bodySize</code> field set to |body size|, <code>content</code>
+   field set to |content|, and the <code>authChallenges</code> field set to
+   |auth challenges| if it's not null, or omitted otherwise.
 
 </div>
 
-### Events ### {#module-network-event}
+#### The network.SetCookieHeader Type #### {#type-network-SetCookieHeader}
 
-#### The network.beforeRequestSent Event #### {#event-network-beforeSendRequest}
-<dl>
-   <dt>Event Type</dt>
-   <dd>
-      <pre class="cddl local-cddl">
-        network.BeforeRequestSent = {
-         method: "network.beforeRequestSent",
-         params: network.BeforeRequestSentParameters
-       }
+[=Remote end definition=]
 
-       network.BeforeRequestSentParameters = {
-         network.BaseParameters,
-         initiator: network.Initiator,
-       }
-      </pre>
-   </dd>
-</dl>
+<pre class="cddl local-cddl remote-cddl">
+network.SetCookieHeader = {
+    name: text,
+    value: network.BytesValue,
+    ? domain: text,
+    ? httpOnly: bool,
+    ? expires: text,
+    ? maxAge: js-uint,
```

Comment:
Minor nitpick: According to [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie), the Max-Age parameter of the header can be negative, so `js-int`? I know that using zero has the same effect as a negative, but to strictly follow the header allowable values here, we should allow it too. I'm not too worried about it either way, if we want to ignore this corner case.